py/selenium/webdriver/remote/webelement.py

```diff
@@ -542,7 +542,12 @@ def rect(self):
         if self._w3c:
             return self._execute(Command.GET_ELEMENT_RECT)['value']
         else:
-            return dict(list(self.size.items()) + list(self.location.items()))
+            size = self._execute(Command.GET_ELEMENT_SIZE)['value']
+            loc = self._execute(Command.GET_ELEMENT_LOCATION)['value']
+
+            rect = {"height": size["height"], "width": size["width"],
+                    "x": round(loc['x']), "y": round(loc['y'])}
+        return rect
```

Comment:
Is there a reason you don't just use `self.size` and `self.location` ? The ending return is outside of the else, which will cause an error if `self._w3c` is true.

Why not something like this?
```
rect = self.size.copy()
rect.upate(self.location)
return rect
```

Reply:
Nested "if" in these methods made me think about a better solution. Your option will work well too

Reply:
But without typo in `rect.upate(self.location)` C:

Reply:
@lmtierney  I made a changes using your example